crates/ruff_linter/src/rules/flake8_pyi/rules/future_annotations_in_stub.rs

```diff
@@ -35,34 +35,38 @@ impl Violation for FutureAnnotationsInStub {
 
 /// PYI044
 pub(crate) fn from_future_import(checker: &mut Checker, target: &StmtImportFrom) {
-    if let StmtImportFrom {
+    let StmtImportFrom {
         range,
         module: Some(name),
         names,
         ..
     } = target
-    {
-        if name == "__future__" && names.iter().any(|alias| &*alias.name == "annotations") {
-            let mut diagnostic = Diagnostic::new(FutureAnnotationsInStub, *range);
+    else {
+        return;
+    };
 
-            if checker.settings.preview.is_enabled() {
-                let stmt = checker.semantic().current_statement();
+    if name != "__future__" || names.iter().all(|alias| &*alias.name != "annotations") {
+        return;
+    };
```

Comment:
Even clearer in my opinion would be

```suggestion
    if name != "__future__" {
        return;
    };
    
    if names.iter().all(|alias| &*alias.name != "annotations") {
        return;
    }
```